rr_bot/main.py

```diff
@@ -35,10 +36,73 @@ def row_generator(args):
     return row_gen
 
 
+def parse_peal_speed(peal_speed: str):
+    """
+    Parses a peal speed written in the format /2h58(m?)/ or /XXX(m?)/ into a number of minutes.
+    """
+
+    def exit_with_message(error_text):
+        """ Print a helpful error message and exit the program with an error code. """
+
+        sys.exit(f"Error parsing peal speed '{peal_speed}': {error_text}")
+
+    # Remove the 'm' from the end of the peal speed - it doesn't add any clarity
+    stripped_peal_speed = peal_speed[:]
```

Comment:
An alternative for slightly more flexibility on the input format (without lots of `if` statements) would be to use something like this pair of regexes.

```
^(?P<hour>\d*)\s*(?:hour|hr|h)\s*(?P<min>\d*)?\s*(?:minutes|min|m)?$
^(?P<min>\d*)\s*(?:minutes|min|m)?$
```

You have good examples of the format in the help text and actionable error messages, so possibly more readable not to change it.

Reply:
The regexes would probably be more compact and flexible, but the current code can track down exactly what the user did wrong, so I think it's fine (also regexes are just difficult to read and document, as well as being very easy to mess up).